doc/mk_api_doc.py

```diff
@@ -47,6 +49,11 @@ def parse_options():
         default=False,
         help='Include ML/OCaml API documentation'
     )
+    parser.add_argument('--js',
```

Comment:
was this tested?
I am trying to reproduce it for the documentation build. See #6155.
At this point trying to produce the --js documentation on top of an emsdk build (not sure if this dependency is required).

An alternative approach may be to decentralize the documentation build and include it in the build-wasm GitHub action to produce an artifact (and wishful thinking to copy it automatically to z3prover.github.io, though I have myself been entirely unsuccessful in getting any pre-cooked action to work properly).

Reply:
@NikolajBjorner I did test it beforehand, yes.

The emsdk is not required to build the documentation, and as you've noticed - it's actually a detriment due to shadowing of Node version. Running separate from emsdk build would be easiest, but if you want to do everything inside one build, the documentation should be built before `emsdk activate` is called. Same way the Typescript is built before emsdk in the [Github Action](https://github.com/Z3Prover/z3/blob/527914db05d80da4d0a070485ddd87720f514c66/.github/workflows/wasm.yml#L38)

@bakkot has written a correct solution to the second problem in #6155 - currently there is an unwritten expectation that the developer has run `npm install` in `src/api/js` directory. `npm --prefix=src/api/js ci` also works and is preferred. Another solution is to quick fail inside the `mk_api_doc.py` script by checking whether `src/api/js/node_modules` folder exists (it's added during `npm install`).

Reply:
OK, thanks it is getting closer now.
Currently some path discrepancy: the relative path used in jest.ts doesn't work with the setup in mk_api_doc.py.